### ISSUE TITLE
hardlink creation fails when /tmp is on a different partition

### ISSUE THREAD

**Issue by maxadamo** [xorpaul/g10k#37]
hardlink cannot cross partitions, and /tmp/g10k is often on a different partition 
It would be handy to have a chance to set cachedir when using puppetfile

**Comment by xorpaul** [comment]
Right now you can set it via an environment variable called `g10k_cachedir`

But you are right it would be nice to set it in the Puppetfile directly.

**Comment by maxadamo** [comment]
I could try to use this variable.... I was expecting an option such as _-cachedir_ 
thanks, anyway.

**Comment by xorpaul** [comment]
Closed with #49

**Comment by maxadamo** [comment]
I leave it close, but I was wondering if you've added a check for the partition. 
With Python I have something like this in place: 
```python
if hex(os.stat(BASE_DIR)[2]) != hex(os.stat(CACHEDIR)[2]):
    print some message and say bye bye
```
we are giving you looot of work to do :)

**Comment by xorpaul** [comment]
Hmm, I didn't find anything like that after searching for golang filesystem partition and thought a check is too complicated.

I'll give it another look :)

**Comment by xorpaul** [comment]
added check for cross-device hardlinking https://github.com/xorpaul/g10k/commit/945e33dfd8e6af9ddd47586f7d0ee7615dfe2f01

Will be released in 0.3
